pkg/support/dump_test.go

```diff
@@ -38,3 +100,279 @@ func TestParseFileName(t *testing.T) {
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			err := dumpNetworkPolicyResources(fs, exe, tc.basedir)
+			defer func() {
+				exe = exec.New()
+			}()
+			if tc.expectedErr != "" {
+				if runtime.IsWindowsPlatform() {
+					require.EqualError(t, err, tc.expectedWinErr)
+				} else {
+					require.EqualError(t, err, tc.expectedErr)
+				}
+			} else {
+				require.NoError(t, err, "Successfully writing to the file")
+			}
+		})
+	}
+}
+
+func TestDumpControllerInfo(t *testing.T) {
+	tests := []struct {
+		name           string
+		basedir        string
+		since          string
+		expectedErr    string
+		expectedWinErr string
+	}{
+		{
+			name:    "controller info specified ",
+			basedir: "dir1/dir2",
+			since:   "5s",
+		},
+		{
+			name:           "controller info not specified",
+			basedir:        "dir1/dir2",
+			since:          "8s",
+			expectedErr:    "error when dumping controllerinfo: executable file not found in $PATH",
+			expectedWinErr: "error when dumping controllerinfo: executable file not found in %PATH%",
+		},
+	}
+
+	exe = new(testExecuter)
+	fs := afero.NewOsFs()
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			varr := NewControllerDumper(fs, exe, tc.since)
+			err := varr.DumpControllerInfo(tc.basedir)
+			defer func() {
+				exe = exec.New()
+			}()
+			if tc.expectedErr != "" {
+				if runtime.IsWindowsPlatform() {
+					require.EqualError(t, err, tc.expectedWinErr)
+				} else {
+					require.EqualError(t, err, tc.expectedErr)
+				}
+			} else {
+				require.NoError(t, err)
+			}
+		})
+	}
+}
+
+func TestControllerDumpNetworkPolicyResources(t *testing.T) {
+	tests := []struct {
+		name           string
+		basedir        string
+		since          string
+		expectedErr    string
+		expectedWinErr string
+	}{
+		{
+			name:    "networkpolicy info specified ",
+			basedir: "dir1/dir2",
+			since:   "5s",
+		},
+		{
+			name:           "networkpolicy info not specified",
+			basedir:        "dir1/dir2",
+			since:          "8s",
+			expectedErr:    "error when dumping networkpolicies: executable file not found in $PATH",
+			expectedWinErr: "error when dumping networkpolicies: executable file not found in %PATH%",
+		},
+	}
+
+	exe = new(testExecuter)
+	fs := afero.NewOsFs()
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			varr := NewControllerDumper(fs, exe, tc.since)
+			err := varr.DumpNetworkPolicyResources(tc.basedir)
+			defer func() {
+				exe = exec.New()
+			}()
+			if tc.expectedErr != "" {
+				if runtime.IsWindowsPlatform() {
+					require.EqualError(t, err, tc.expectedWinErr)
+				} else {
+					require.EqualError(t, err, tc.expectedErr)
+				}
+			} else {
+				require.NoError(t, err)
+			}
+		})
+	}
+}
+
+func TestControllerDumpHeapPprof(t *testing.T) {
+	basedir := "dir1/dir2"
+	since := "5s"
+	expectedErr := ""
+	expectedWinErr := "error when creating file  to write : open : The system cannot find the file specified."
```

Comment:
why does this fail on Windows but not Linux? Is it the expected behavior, can you add a comment?